cmd/embedded-cluster/proxy.go

```diff
@@ -51,17 +56,26 @@ func getProxySpecFromFlags(c *cli.Context) *ecv1beta1.ProxySpec {
 		proxy.HTTPSProxy = c.String("https-proxy")
 	}
 	if c.String("no-proxy") != "" {
-		noProxy = append(noProxy, c.String("no-proxy"))
+		providedNoProxy = append(providedNoProxy, c.String("no-proxy"))
 	}
+	proxy.ProvidedNoProxy = strings.Join(providedNoProxy, ",")
+	combineNoProxySuppliedValuesAndDefaults(c, proxy)
+	if proxy.HTTPProxy == "" && proxy.HTTPSProxy == "" && proxy.NoProxy == "" {
+		return nil
+	}
+	return proxy
+}
+
+func combineNoProxySuppliedValuesAndDefaults(c *cli.Context, proxy *ecv1beta1.ProxySpec) {
```

Comment:
name is too long 😆, but i guess it's fine. i would've gone with something like `injectNoProxyDefaults` or something.
```suggestion
func combineNoProxySuppliedValuesAndDefaults(c *cli.Context, proxy *ecv1beta1.ProxySpec) {
```